frame/staking/src/pallet/impls.rs

```diff
@@ -1681,12 +1681,6 @@ impl<T: Config> StakingInterface for Pallet<T> {
 #[cfg(any(test, feature = "try-runtime"))]
 impl<T: Config> Pallet<T> {
 	pub(crate) fn do_try_state(_: BlockNumberFor<T>) -> Result<(), &'static str> {
-		ensure!(
-			T::VoterList::iter()
-				.all(|x| <Nominators<T>>::contains_key(&x) || <Validators<T>>::contains_key(&x)),
```

Comment:
Could you still try what difference it makes if you read the whole maps first and then iterate locally?  
This is probably slow since it has quadratic storage complexity.  
You can load the whole map with `map::iter().collect()`.

Reply:
Good point, will do!

Reply:
So this was totally wrong. I was not copying session keys which is why the try state was running fast for me locally.
The voterlist tests were still fast probably because second reads are from cache.

The test that is taking long time is the `check_nominators`. It has cubic time complexity (Total Nominators * Active Validator * Validator exposure list).

Reply:
Moving to draft till I find some way to speed up the above.